client/proposal.go

```diff
@@ -356,6 +365,38 @@ func (c *Client) validSubChannelProposal(proposal *SubChannelProposal) error {
 	return nil
 }
 
+func (c *Client) validVirtualChannelProposal(prop *VirtualChannelProposal, ourIdx channel.Index) error {
+	numParents := len(prop.Parents)
+	numPeers := prop.NumPeers()
+	if numParents != numPeers {
+		return errors.Errorf("expected %d parent channels, got %d", numPeers, numParents)
+	}
+
+	parent, err := c.Channel(prop.Parents[ourIdx])
+	if err != nil {
+		return errors.New("parent channel not found")
+	}
+
+	if err := channel.AssetsAssertEqual(parent.State().Assets, prop.InitBals.Assets); err != nil {
+		return errors.WithMessage(err, "unequal assets")
+	}
+
+	if !prop.InitBals.Balances.Equal(prop.FundingAgreement) {
+		return errors.WithMessage(err, "unequal funding agreement")
+	}
+
+	numIndexMaps := len(prop.IndexMaps)
+	if numIndexMaps != numPeers {
+		return errors.Errorf("expected %d index maps, got %d", numPeers, numIndexMaps)
+	}
+
+	virtualBals := transformBalances(prop.InitBals.Balances, parent.state().NumParts(), prop.IndexMaps[ourIdx])
```

Comment:
Thanks for catching this one. The parent channel is not locked, but you are right that it would be better if it was. However, ensuring this is a bit tricky. I think we should have call about this.

Reply:
Fixed, please have a look.